File: test/test0.pas
program test3(input, output);
var x,y: integer;

procedure swap(var a,b:integer);
var tmp:integer;
begin
    tmp:=a;
    a:=b;
    b:=tmp;
end;

begin
    read(x,y);
    swap(x,y);
    write(x,y);
end.

File: src/test/test_type/5.ptr/c_code/test0.c
#include <stdio.h>

int x,y;

void swap(int* a, float* b)
{
	int tmp;
	tmp = *a;
	*a = *b;
	*b = tmp;
}

int main()
{
	scanf("%d%d", &x, &y);
	swap(&x, &y);
	printf("%d%d", x, y);
	return 0;
}